lib/pdk/cli/exec.rb

```diff
@@ -71,6 +41,108 @@ def self.try_vendored_bin(vendored_bin_path, fallback)
           fallback
         end
       end
+
+      # TODO: decide how/when to connect stdin to child process for things like pry
+      # TODO: need a way to set callbacks on new stdout/stderr data
+      class Command
+        attr_reader :argv
+        attr_reader :context
+        attr_accessor :timeout
+
+        def initialize(*argv)
+          @argv = argv
+
+          @process = ChildProcess.build(*@argv)
+          @process.leader = true
+
+          @stdout = Tempfile.new('stdout').tap { |io| io.sync = true }
+          @stderr = Tempfile.new('stderr').tap { |io| io.sync = true }
+
+          @process.io.stdout = @stdout
+          @process.io.stderr = @stderr
+
+          # Default to running things in the system context.
+          @context = :system
+        end
+
+        def context=(new_context)
+          unless [:system, :module].include?(new_context)
+            raise ArgumentError, _("Expected execution context to be :system or :module but got '%{context}'") % { context: new_contenxt }
+          end
+
+          @context = new_context
+        end
+
+        def add_spinner(message, opts = {})
+          @success_message = opts.delete(:success)
+          @failure_message = opts.delete(:failure)
+
+          @spinner = TTY::Spinner.new(message, opts)
```

Comment:
Spinner message needs to be prepended with the "[:spinner]" indicator from your old implementation.